qiskit/opflow/primitive_ops/pauli_sum_op.py

```diff
@@ -442,4 +451,10 @@ def is_zero(self) -> bool:
         return op.coeff == 1 and len(op) == 1 and primitive.coeffs[0] == 0
 
     def is_hermitian(self):
-        return np.isreal(self.coeffs).all() and np.all(self.primitive.paulis.phase == 0)
+        if not self.coeffs.dtype == object:
+            return np.isreal(self.coeffs).all() and np.all(self.primitive.paulis.phase == 0)
+        else:
+            is_real = []
+            for coeff in self.coeffs:
+                is_real.append(np.isreal(coeff))
```

Comment:
There's no need to build up a list here - if any of them are false as you're looping through, you can immediately return `False`.